managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/ManagedCursorImpl.java

```diff
@@ -1817,21 +1809,10 @@ public void operationFailed(ManagedLedgerException exception) {
 
     private void flushPendingMarkDeletes() {
         if (!pendingMarkDeleteOps.isEmpty()) {
-            if (RESET_CURSOR_IN_PROGRESS_UPDATER.get(this) == TRUE) {
-                failPendingMarkDeletes();
-            } else {
-                internalFlushPendingMarkDeletes();
-            }
+            internalFlushPendingMarkDeletes();
```

Comment:
`internalFlushPendingMarkDeletes` just clears previous `pendingMarkDeleteOps` and not completes the callback of `PendingMarkDeleteEntry` and there could be possibility if someone is waiting for the callback. One example [PersistentReplicator](https://github.com/yahoo/pulsar/blob/master/pulsar-broker/src/main/java/com/yahoo/pulsar/broker/service/persistent/PersistentReplicator.java#L596) waiting to just do debug-log and which is fine but just concern is that as it has callback so, one can wait on it..